### ISSUE TITLE
Memorable Date component: Error Message Not Announced or Visible for Future Dates

### ISSUE THREAD

**Issue by jeana-adhoc** [department-of-veterans-affairs/vets-design-system-documentation#4924]
# Bug Report

- [X] I’ve searched for any related issues and avoided creating a duplicate issue.

## What happened

When entering a date in the future using the memorable date component, the component shows a red border on the left (error state), but no error message is visible or announced until the user clicks the continue button. After clicking continue, sometimes an error message appears and focus shifts to the component. In other cases, the error message does not appear visually at all.

## What I expected to happen

Since, the red border is appearing onblur (assuming on validation) I expected the error message to also appear visually (when entering an invalid date), so users are immediately notified of the issue.

OR

I expected no visual response to onblur whatsoever, and expected the error UI treatment AND error message to appear only after clicking "continue"

## Reproducing

- CSS Library version: 
- Device: 
- Browser: 

**URL(s) where this bug occurs**

Where bug was discovered:
- https://staging.va.gov/disability/eligibility/special-claims/unemployability/employment-questionnaire-form-21-4140/employers/1/employment-dates?add=true

Where anyone can see the bug without logging in to VA.gov (And it confirms that this is live on production):
- https://www.va.gov/burials-memorials/memorial-items/presidential-memorial-certificates/request-certificate-form-40-0247/veteran-personal-information

### Steps to reproduce

**Scenario 1:**
- Navigate to https://staging.va.gov/disability/eligibility/special-claims/unemployability/employment-questionnaire-form-21-4140/employers/1/employment-dates?add=true
- Enter a date in the future into memorable date
- Observe red border appearing, but no error message appearing or being announced
- Click continue
- Error message appears, focus is taken to the entire component, but error message is never announced (similar issues: [#114355](https://github.com/department-of-veterans-affairs/va.gov-team/issues/114355))

https://github.com/user-attachments/assets/ff50406d-0c74-4e58-8d66-ca6b61b58df5

The fix here needs to be on making sure the message appears visually and is in the dom. Making sure it is announced would be another effort connected to all error message on all form inputs.

**Scenario 2**

- Navigate to https://staging.va.gov/disability/eligibility/special-claims/unemployability/employment-questionnaire-form-21-4140/employers/1/employment-dates?add=true
- Enter a date in the future into memorable date
- Observe the red border appearing
- With screen reader keyboard controls I traversed through the enitre component to see if maybe there was an error message that would be announced by the screen reader, and there was not.
- When I clicked continue, focus goes *back* to the component, and unlike the other scenario, *no* error message visually appears either.

https://github.com/user-attachments/assets/85f29c59-75e8-4185-864a-12c7dff9e4a2

**To reproduce without logging in**

1. Go to this form [Presidential Memorial Certificate](https://www.va.gov/burials-memorials/memorial-items/presidential-memorial-certificates/request-certificate-form-40-0247)  .
2. Start the form
3. On the first form page, enter a date in the future into the memorable date field.
4. Tab past the year field
5. Observe that a red border appears, but no error message is shown or announced.
6. This publicly available form is a little different in that it has 2 date fields. Once you enter a date in the other field, and press tab the previous fields error message appears. On other forms where there is only one date field on the page, pressing continue has the same affect.

## How urgent is this request? 

Please select the appropriate option below and/or provide details

- [X] This bug is blocking work currently in progress
- [ ] This bug is affecting work currently in progress but we have a workaround
- [ ] This bug is blocking work planned within the next few sprints
- [ ] This bug is not blocking any work
- [ ] Other

I think this could rise to an OOO deployment fix as with no error message appearing, it may be difficult for people to know what is wrong with their date submission.

### Details

This bug affects both visual and assistive technology users, causing confusion and failing accessibility requirements for error handling and screen reader support.

References:
- WCAG 3.3.1 Error Identification
- WCAG 4.1.2 Name, Role, Value
